src/file_utils.rs

```diff
@@ -235,3 +237,51 @@ pub(crate) fn add_recert_edited_annotation(_resource: &mut Value, _yaml_location
 
     Ok(())
 }
+
+fn time_rfc3339(asn1time: &x509_certificate::asn1time::Time) -> String {
+    match asn1time {
+        x509_certificate::asn1time::Time::UtcTime(time) => time.to_rfc3339_opts(SecondsFormat::Secs, true),
+        x509_certificate::asn1time::Time::GeneralTime(time) => {
+            DateTime::<Utc>::from((*time).clone()).to_rfc3339_opts(SecondsFormat::Secs, true)
+        }
+    }
+}
+
+/// Updates the auth.openshift.io/certificate-not-{after,before} annotations to match the
+/// validity period of the regenerated certificate. When such annotations are missing, it skips
+/// them. Those annotations are used by cluster operators based on library-go to rotate those crypto
+/// objects via the certrotation component.
+///
+/// Reference:
+/// - https://github.com/openshift/library-go/blob/master/pkg/operator/certrotation/signer.go#L85
+pub(crate) fn update_auth_certificate_annotations(resource: &mut Value, certificate: &Certificate) -> Result<()> {
+    if resource.pointer_mut("/metadata/annotations").is_none() {
+        return Ok(());
+    }
+
+    let annotations = resource
+        .pointer_mut("/metadata/annotations")
+        .context("annotations must exist")?
+        .as_object_mut()
+        .context("annotations must be an object")?;
+
+    if !annotations.contains_key("auth.openshift.io/certificate-not-after")
+        || !annotations.contains_key("auth.openshift.io/certificate-not-before")
+    {
+        return Ok(());
+    }
+
+    let cert: &x509_certificate::X509Certificate = &certificate.cert;
+    let certificate: &x509_certificate::rfc5280::Certificate = cert.as_ref();
+
+    annotations.insert(
+        String::from("auth.openshift.io/certificate-not-after"),
+        Value::String(time_rfc3339(&certificate.tbs_certificate.validity.not_after)),
+    );
+    annotations.insert(
+        String::from("auth.openshift.io/certificate-not-before"),
+        Value::String(time_rfc3339(&certificate.tbs_certificate.validity.not_before)),
+    );
+
+    Ok(())
+}
```

Comment:
A bit simplified implementation (untested)

```suggestion
pub(crate) fn update_auth_certificate_annotations(resource: &mut Value, certificate: &Certificate) -> Result<()> {
    if let Some(annotations) = resource.pointer_mut("/metadata/annotations") {
        let annotations = annotations.as_object_mut().context("annotations must be an object")?;

        let cert: &x509_certificate::X509Certificate = &certificate.cert;
        let certificate: &x509_certificate::rfc5280::Certificate = cert.as_ref();

        if let Some(not_before) = annotations.get_mut("auth.openshift.io/certificate-not-before") {
            *not_before = Value::String(time_rfc3339(&certificate.tbs_certificate.validity.not_before));
        }

        if let Some(not_after) = annotations.get_mut("auth.openshift.io/certificate-not-after") {
            *not_after = Value::String(time_rfc3339(&certificate.tbs_certificate.validity.not_after));
        }
    }

    Ok(())
}
```

Reply:
Niiiiiice, testing it now

Reply:
I think it's not exactly the same (the previous implementation was only changing the annotations when both {after,before} were present), but it works perfectly, it's idiomatic and awesomely more readable. So let's go with that :+1: